### PR TITLE
Add alt text to scikit-learn documentation

### PR DESCRIPTION
**Pull request by isabela-pf**
# Welcome!

This is the working PR for the PyLadies SWFL, PyLadies Miami, and Python SWFL alt text mini sprint (that's a lot of words)! This is where we collaborate during the event before we send a clean contribution over to the main repo.

## How to contribute

For the full info about this event, please visit the [HackMD agenda](https://hackmd.io/2uOMXV_fT0ycLyo4uH67HQ).

During the event, you can contribute alt text to any of the images on currently used in a blog post. Please contribute via [suggestions](https://hackmd.io/@marslee/BksnYfWVK) so we can easily review each other's work.

If you plan to work on a file, let people know by commenting below. We can have multiple suggestions for the same alt text, but it would be nice if we spread out to cover more images.

If you think an image is missing from these files, let @isabela-pf know and she'll add it to the PR.

## Images in this PR

To make collaborating easier, comment below with the number of the image you'd like to work on. You can do this as a group, and you can write alt text for as many images as you'd like. 

File: `doc/modules/biclustering.rst` ([Link to biclustering docs and rendered images](https://scikit-learn.org/stable/modules/biclustering.html))

- [x] 1. Line 51 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_spectral_coclustering_003.png))
- [x] 2. Line 64 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_spectral_biclustering_003.png))

File: `doc/modules/calibration.rst` ([Link to the calibration docs and rendered images](https://scikit-learn.org/stable/modules/calibration.html))

- [x] 3. Line 56 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_compare_calibration_001.png))

File: `doc/modules/clustering.rst` ([Link to the clustering docs and rendered images](https://scikit-learn.org/stable/modules/clustering.html))

- [ ] 4. Line 36 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_cluster_comparison_001.png))
- [ ] 5. Line 170 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_kmeans_assumptions_001.png))
- [ ] 6. Line 187 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_kmeans_digits_001.png))
- [x] 7. Line 278 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_mini_batch_kmeans_001.png))
- [x] 8. Line 317 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_affinity_propagation_001.png))
- [ ] 9. Line 427 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_mean_shift_001.png))
- [x] 10. Line 470 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_segmentation_toy_001.png))
- [x] 11. Line 475 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_segmentation_toy_002.png))
- [ ] 12. Line 503 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_coin_segmentation_001.png))
- [ ] 13. Line 508 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_coin_segmentation_002.png))
- [ ] 14. Line 611 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_linkage_comparison_001.png))
- [ ] 15. Line 637 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_agglomerative_dendrogram_001.png))
- [x] 16. Line 656 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_ward_structured_vs_unstructured_001.png))
- [x] 17. Line 661 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_ward_structured_vs_unstructured_002.png))
- [ ] 18. Line 711 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_agglomerative_clustering_001.png))
- [ ] 19. Line 716 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_agglomerative_clustering_002.png))
- [ ] 20. Line 721 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_agglomerative_clustering_003.png))
- [ ] 21. Line 726 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_agglomerative_clustering_004.png))
- [ ] 22. Line 751 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_agglomerative_clustering_metrics_005.png))
- [ ] 23. Line 756 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_agglomerative_clustering_metrics_006.png))
- [ ] 24. Line 761 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_agglomerative_clustering_metrics_007.png))
- [x] 25. Line 822 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_dbscan_001.png))
- [ ] 26. Line 908 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_optics_001.png))
- [ ] 27. Line 1051 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_birch_vs_minibatchkmeans_001.png))
- [ ] 28. Line 1543 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_adjusted_for_chance_measures_001.png))

File: `doc/modules/covariance.rst` [(Link to the covariance docs and rendered images](https://scikit-learn.org/stable/modules/covariance.html))

- [x] 29. Line 157 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_covariance_estimation_001.png))
- [x] 30. Line 182 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_lw_vs_oas_001.png))
- [x] 31. Line 215 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_sparse_cov_001.png))
- [x] 32. Line 339 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_robust_vs_empirical_covariance_001.png))
- [x] 33. Line 344 ([image](https://scikit-learn.org/stable/_images/sphx_glr_plot_mahalanobis_distances_001.png))

---

If you need anything, please ask @isabela-pf or @marsbarlee! 🌻

### CODE REVIEW

**Comment by isabela-pf** [comment]
Here's a checklist of a few formatting factors you can check for when reviewing alt text:

The alt text has
- [ ] Correct spelling and no typos
- [ ] Periods and commas where relevant
- [ ] No more than three short, complete sentences
- [ ] A logical way of fitting in the rest of the documentation
- [ ] Consistent text and/or descriptions for the same elements in different images
- [ ] A description of any text in the image

**Comment by MarsBarLee** [comment]
The [link to my slides from earlier](https://docs.google.com/presentation/d/1PTYVwcu2U6wTY-ozrKEK981CEp79MkBtRJymr2LQgCY/edit?usp=sharing)


[Step by Step: How to Add Alt-text with Github Suggestions](https://hackmd.io/@marslee/BksnYfWVK) (example image below)
image

File: doc/modules/biclustering.rst [hunk 1]
```diff
@@ -48,6 +48,7 @@ columns:
    :target: ../auto_examples/bicluster/images/sphx_glr_plot_spectral_coclustering_003.png
    :align: center
    :scale: 50
+   :alt:
```
Comment (isabela-pf) [inline doc/modules/biclustering.rst]:
```suggestion
   :alt: This is an example!
```

Comment (reshamas) [reply doc/modules/biclustering.rst]:
```suggestion
   :alt: The graph is a square heat map, 5x5, with axes from 0 to 250.  The darkest 5 squares of heat map run diagonally from top left to bottom right.
```

File: doc/modules/clustering.rst [hunk 2]
```diff
@@ -272,6 +275,7 @@ small, as shown in the example and cited reference.
    :target: ../auto_examples/cluster/plot_mini_batch_kmeans.html
    :align: center
    :scale: 100
+   :alt:
```
Comment (bsipocz) [inline doc/modules/clustering.rst]:
```suggestion
   :alt: A figure of three panels with scatter plots for KMeans, MiniBatchKMeans,
        and their difference.  Both methods have very similar results, the left and
        middle panels show the three identified clusters.  The difference
        panel highlights the <20 points that differ out of the 3000 points. 
```

File: doc/modules/clustering.rst [hunk 3]
```diff
@@ -798,6 +819,7 @@ by black points below.
 .. |dbscan_results| image:: ../auto_examples/cluster/images/sphx_glr_plot_dbscan_001.png
         :target: ../auto_examples/cluster/plot_dbscan.html
         :scale: 50
+        :alt:
```
Comment (sunsyray) [inline doc/modules/clustering.rst]:
```suggestion
        :alt: A bubble chart that maps three clusters, with the x-axis from -2.5 to 2 and the y-axis from -2 to 2.5. Outliers are indicated separately and are evenly spaced around the clusters.
```

File: doc/modules/biclustering.rst [hunk 4]
```diff
@@ -60,6 +61,7 @@ small:
    :target: ../auto_examples/bicluster/images/sphx_glr_plot_spectral_biclustering_003.png
    :align: center
    :scale: 50
+   :alt:
```
Comment (Madelene) [inline doc/modules/biclustering.rst:64]: The graph is a square heat map, 5x5, with axes from 0 to 250. The variance of the values within each bicluster is small.

Comment (Madelene) [inline doc/modules/biclustering.rst]:
```suggestion
   :alt: The graph is a square heat map, 5x5, with axes from 0 to 250. The variance of the values within each bicluster is small, causing a checkerboard effect.
```

File: doc/modules/clustering.rst [hunk 5]
```diff
@@ -461,10 +467,12 @@ computed using a function of a gradient of the image.
 .. |noisy_img| image:: ../auto_examples/cluster/images/sphx_glr_plot_segmentation_toy_001.png
     :target: ../auto_examples/cluster/plot_segmentation_toy.html
     :scale: 50
+    :alt:
 
 .. |segmented_img| image:: ../auto_examples/cluster/images/sphx_glr_plot_segmentation_toy_002.png
     :target: ../auto_examples/cluster/plot_segmentation_toy.html
     :scale: 50
+    :alt:
```
Comment (InessaPawson) [inline doc/modules/clustering.rst]:
```suggestion
    :alt: An image of connected nearly equally sized circles. Each circle has a distinct color: dark green, light green, blue, and yellow. The clarity of the image is much better compared to the previous one.
```

File: doc/modules/clustering.rst [hunk 6]
```diff
@@ -461,10 +467,12 @@ computed using a function of a gradient of the image.
 .. |noisy_img| image:: ../auto_examples/cluster/images/sphx_glr_plot_segmentation_toy_001.png
     :target: ../auto_examples/cluster/plot_segmentation_toy.html
     :scale: 50
+    :alt:
```
Comment (InessaPawson) [inline doc/modules/clustering.rst]:
```suggestion
    :alt: An image of connected nearly equally sized circles. The circles are in shades of green, blue, and yellow. The entire image is pixelated, blurry.
```

File: doc/modules/clustering.rst [hunk 7]
```diff
@@ -641,10 +653,12 @@ the roll.
 .. |unstructured| image:: ../auto_examples/cluster/images/sphx_glr_plot_ward_structured_vs_unstructured_001.png
         :target: ../auto_examples/cluster/plot_ward_structured_vs_unstructured.html
         :scale: 49
+        :alt:
```
Comment (bsipocz) [inline doc/modules/clustering.rst]:
```suggestion
        :alt: A 3D scatter plot of 1500 data points. Points are located along a swirl
             or swiss roll. The six clusters are highlighted in different colours. 
             The hierarchical clustering is performed without connectivity
             constraints on the structure and is solely based on distance therefore
             clusters extend to multiple layers of the roll.
```

File: doc/modules/covariance.rst [hunk 8]
```diff
@@ -210,6 +212,7 @@ cross-validation to automatically set the ``alpha`` parameter.
    :target: ../auto_examples/covariance/plot_sparse_cov.html
    :align: center
    :scale: 60%
+   :alt:
```
Comment (MarsBarLee) [inline doc/modules/covariance.rst]:
```suggestion
   :alt: Various matrix using different estimators: Empircal covariance, Empircal precision, Ledoit-Wolf covariance, Ledoit-Wolf precision, GraphicaLassoCV covariance, GraphicaLassoCV precision, True covariance and True Precision.

```

File: doc/modules/clustering.rst [hunk 9]
```diff
@@ -641,10 +653,12 @@ the roll.
...
 
 .. |structured| image:: ../auto_examples/cluster/images/sphx_glr_plot_ward_structured_vs_unstructured_002.png
         :target: ../auto_examples/cluster/plot_ward_structured_vs_unstructured.html
         :scale: 49
+        :alt:
```
Comment (bsipocz) [inline doc/modules/clustering.rst]:
```suggestion
        :alt: A 3D scatter plot of 1500 data points. Points are located along a swirl
              or swiss roll. The six clusters are highlighted in different colours.
              The hierarchical clustering is performed with connectivity constraints
              that respect the structure of the roll therefore the clusters form a nice
              parcellation along the roll.
```

File: doc/modules/clustering.rst [hunk 10]
```diff
@@ -310,6 +314,7 @@ is given.
    :target: ../auto_examples/cluster/plot_affinity_propagation.html
    :align: center
    :scale: 50
+   :alt:
```
Comment (isabela-pf) [inline doc/modules/clustering.rst]:
```suggestion
   :alt: Three distinct clusters created using affinity propagation.
```

by @reshamas

File: doc/modules/covariance.rst [hunk 11]
```diff
@@ -178,6 +179,7 @@ object to the same sample.
    :target: ../auto_examples/covariance/plot_lw_vs_oas.html
    :align: center
    :scale: 75%
+   :alt:
```
Comment (aguspesce) [inline doc/modules/covariance.rst]:
```suggestion
   :alt: Two figures. One of them plots the Mean Squared Error difference between a LedoitWolf and an OAS estimator of the covariance with the y-axis from 0 to 60.  The other one compares the Shrinkage covariance estimation between the  LedoitWolf  and OAS, with the  y-axis from 0 to 1. In iboth figures the x-axis is the number of samples from 5 to 30.
   
```

**Comment by isabela-pf** [comment]
Hey everyone! Just a reminder that I will close this PR soon to get it ready for submitting to the main repo. I'm planning to clean everything up and reply to the main issue in the next 24–48 hours, so it'd be best if you make any final submissions before then. I know I still want to write some 😆.

File: doc/modules/covariance.rst [hunk 12]
```diff
@@ -333,10 +336,12 @@ attributes of a :class:`MinCovDet` robust covariance estimator object.
 .. |robust_vs_emp| image:: ../auto_examples/covariance/images/sphx_glr_plot_robust_vs_empirical_covariance_001.png
    :target: ../auto_examples/covariance/plot_robust_vs_empirical_covariance.html
    :scale: 49%
+   :alt:
```
Comment (aerikpawson) [inline doc/modules/covariance.rst:339]: :alt: Two line charts comparing the estimation errors that are made when using various types of location and covariance estimates on contaminated Gaussian distributed data sets.

Comment (aerikpawson) [inline doc/modules/covariance.rst]:
```suggestion
   :alt: Two line charts comparing estimation errors that are made when using various types of location and covariance estimates on contaminated Gaussian distributed data sets.
```

File: doc/modules/calibration.rst [hunk 13]
```diff
@@ -53,6 +53,7 @@ by showing the number of samples in each predicted probability bin.
 .. figure:: ../auto_examples/calibration/images/sphx_glr_plot_compare_calibration_001.png
    :target: ../auto_examples/calibration/plot_compare_calibration.html
    :align: center
+   :alt:
```
Comment (isabela-pf) [inline doc/modules/calibration.rst]:
```suggestion
   :alt: Five plots comparing calibration classifiers. The first plot compares Logistic, Naive Bayes, SVC, and Random forest classifiers side by side with perfect calibration. The following four histograms isolate each classifier to describe their predicted probability on the x-axis and count on the y-axis.
```

File: doc/modules/clustering.rst (hunk 3 above)
Comment (isabela-pf) [reply doc/modules/clustering.rst]: Just so you know, I only edited this comment to be a suggestion so you could get commit credit for writing it. Thanks for your work!

**Comment by isabela-pf** [comment]
Again, thank you to everyone who participated. I hope this was a good exercise for you (and maybe inspires you to write alt text on other projects)! 

I'm closing this PR to open one on the proper scikit-learn repo and will link it here once the PR is up.

**Comment by isabela-pf** [comment]
It's PR time! Look at all you cool kids in the commit history! :sunflower: https://github.com/scikit-learn/scikit-learn/pull/21354

Thanks again for taking the time to contribute.

**Pull request closed without merging**